markymark/Classes/Rules/Inline/ShortLinkRule.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ShortLinkRule.swift
+//  markymark
+//
+//  Created by Jim van Zummeren on 20/07/2021.
+//  Copyright © 2021 M2mobi. All rights reserved.
+//
+
+import Foundation
+
+open class ShortLinkRule: InlineRule {
+    
+    /// Example: <http://www.google.com>
+    open var expression = NSRegularExpression.expressionWithPattern(
+        #"<(.+?)>"#
+    )
+        
+    public func getAllMatches(_ lines: [String]) -> [NSRange] {
+        guard let line = lines.first else { return [] }
+        return getResults(line).map { $0.range }
+    }
+
+    // MARK: Rule
+
+    public func recognizesLines(_ lines: [String]) -> Bool {
+        guard let line = lines.first else { return false }
+        return getResults(line).count > 0
+    }
+
+    public func createMarkDownItemWithLines(_ lines: [String]) -> MarkDownItem {
+        let url: String? = lines.first?.subStringWithExpression(expression, ofGroup: 1)
+
+        return LinkMarkDownItem(
+            lines: lines,
+            content: url ?? "",
+            url: url ?? ""
+        )
+    }
+}
+
+private extension ShortLinkRule {
+
+    private func getResults(_ string: String) -> [NSTextCheckingResult] {
```

Comment:
I thought xcode wouldn't say these private annotations are redundant as the extension itself is already private.

Reply:
Solved